integration/kube_integration_test.go

```
@@ -0,0 +1,535 @@
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/testauthority"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/events"
+	kubeutils "github.com/gravitational/teleport/lib/kube/utils"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/session"
+	"github.com/gravitational/teleport/lib/utils"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/check.v1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	streamspdy "k8s.io/apimachinery/pkg/util/httpstream/spdy"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/client-go/transport/spdy"
+)
+
+var _ = check.Suite(&KubeSuite{})
+
+type KubeSuite struct {
+	*kubernetes.Clientset
+	kubeConfig *rest.Config
+	ports      utils.PortList
+	me         *user.User
+	// priv/pub pair to avoid re-generating it
+	priv []byte
+	pub  []byte
+
+	// kubeCACert is a certificate of a kubernetes certificate authority
+	kubeCACert []byte
+	// kubeCACertPath is a temp file to hold CA certificate authority contents
+	kubeCACertPath string
+	// kubeAPIHost is a host:port with kubernetes API server
+	kubeAPIHost string
+}
+
+func (s *KubeSuite) SetUpSuite(c *check.C) {
+	var err error
+	utils.InitLoggerForTests()
+	SetTestTimeouts(time.Millisecond * time.Duration(100))
+
+	s.priv, s.pub, err = testauthority.New().GenerateKeyPair("")
+	c.Assert(err, check.IsNil)
+
+	s.ports, err = utils.GetFreeTCPPorts(AllocatePortsNum, utils.PortStartingNumber+AllocatePortsNum+1)
+	if err != nil {
+		c.Fatal(err)
+	}
+	s.me, _ = user.Current()
```

Do not ignore the error?